internal/testutil/reindent_test.go

```diff
@@ -70,3 +70,44 @@ func (*S) testReindent(c *C, test reindentTest) {
 	}
 	c.Assert(string(reindented), Equals, test.result)
 }
+
+type indentLinesTest struct {
+	raw, prefix, result, error string
+}
+
+var indentLinesTests = []indentLinesTest{{
+	raw:    "a\n\tb\n  \t\tc\td\n\t ",
+	prefix: "foo",
+	result: "fooa\nfoo\tb\nfoo  \t\tc\td\nfoo\t ",
+}}
+
+func (s *S) TestIndentLines(c *C) {
+	for _, test := range indentLinesTests {
+		s.testIndentLines(c, test)
+	}
+}
+
+func (s *S) testIndentLines(c *C, test indentLinesTest) {
+	defer func() {
+		if err := recover(); err != nil {
+			errMsg, ok := err.(string)
+			if !ok {
+				panic(err)
+			}
+			c.Assert(errMsg, Equals, test.error)
+		}
+	}()
+
+	c.Logf("Test: %#v", test)
+
+	if !strings.HasSuffix(test.result, "\n") {
+		test.result += "\n"
+	}
+
+	indented := testutil.IndentLines(test.raw, test.prefix)
+	if test.error != "" {
+		c.Errorf("Expected panic with message '%#v'", test.error)
+		return
+	}
+	c.Assert(string(indented), Equals, test.result)
+}
```

Comment:
I think that this function has to be simplified. For instance, because there is no point in us testing the error in `WriteString`, I think we should remove all the logic for handling the panic. If the test panics it will fail automatically.

This bit here is also a little bit confusing:
```
	if !strings.HasSuffix(test.result, "\n") {
		test.result += "\n"
	}
```
IMO the test result should be exactly what we expect to return, we should not have to read the implementation to interpret the return.

Reply:
Refactored this. Thanks a bunch.